packages/api/src/protocols/zeroex-v4/swap-token.ts

```diff
@@ -0,0 +1,25 @@
+import { Logic } from 'src/types';
+import * as common from '@protocolink/common';
+import { getProtocolTokenList, quote } from 'src/api';
+import * as logics from '@protocolink/logics';
+
+export type SwapTokenParams = common.Declasifying<logics.zeroexv4.SwapTokenLogicParams>;
+
+export type SwapTokenFields = common.Declasifying<logics.zeroexv4.SwapTokenLogicFields>;
+
+export type SwapTokenLogic = Logic<SwapTokenFields>;
+
+export async function getSwapTokenTokenList(chainId: number): Promise<common.Token[]> {
```

Comment:
In the PR for protocolink-logics 0x, changes were made so that `paraswapv5` and `zeroexv4` use the common `TokenList` type `common.Token[]`, causing a unit test failure: https://github.com/dinngo/protocolink-js-sdk/actions/runs/8322650598/job/22770807173?pr=42#step:6:3717

To ensure consistency across all logics with a `TokenList` type, I suggest introducing a `logics.zeroexv4.SwapTokenList` type. This way, `getTokenList()` in both `paraswapv5` and `zeroexv4` can return their specific types. What are your thoughts?

Reply:
agreed. will do.